temporal-sdk/src/main/java/io/temporal/client/WorkflowStartOperationUpdate.java

```diff
@@ -0,0 +1,258 @@
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1});
+  }
+
+  public static <T, A1, A2, R> Builder<R> newBuilder(
+      Functions.Func3<T, A1, A2, R> update, A1 arg1, A2 arg2) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1, arg2});
+  }
+
+  public static <T, A1, A2, A3, R> Builder<R> newBuilder(
+      Functions.Func4<T, A1, A2, A3, R> update, A1 arg1, A2 arg2, A3 arg3) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1, arg2, arg3});
+  }
+
+  public static <T, A1, A2, A3, A4, R> Builder<R> newBuilder(
+      Functions.Func5<T, A1, A2, A3, A4, R> update, A1 arg1, A2 arg2, A3 arg3, A4 arg4) {
+    return WorkflowStartOperationUpdate.createBuilder(
+        update, new Object[] {arg1, arg2, arg3, arg4});
+  }
+
+  public static <T, A1, A2, A3, A4, A5, R> Builder<R> newBuilder(
+      Functions.Func6<T, A1, A2, A3, A4, A5, R> update,
+      A1 arg1,
+      A2 arg2,
+      A3 arg3,
+      A4 arg4,
+      A5 arg5) {
+    return WorkflowStartOperationUpdate.createBuilder(
+        update, new Object[] {arg1, arg2, arg3, arg4, arg5});
+  }
+
+  public static <T> Builder<Void> newBuilder(Functions.Proc1<T> update) {
+    SerializedLambda meta = JavaLambdaUtils.toSerializedLambda(update);
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {});
+  }
+
+  public static <T, A1> Builder<Void> newBuilder(Functions.Proc2<T, A1> update, A1 arg1) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1});
+  }
+
+  public static <T, A1, A2> Builder<Void> newBuilder(
+      Functions.Proc3<T, A1, A2> update, A1 arg1, A2 arg2) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1, arg2});
+  }
+
+  public static <T, A1, A2, A3> Builder<Void> newBuilder(
+      Functions.Proc4<T, A1, A2, A3> update, A1 arg1, A2 arg2, A3 arg3) {
+    return WorkflowStartOperationUpdate.createBuilder(update, new Object[] {arg1, arg2, arg3});
+  }
+
+  public static <T, A1, A2, A3, A4> Builder<Void> newBuilder(
+      Functions.Proc5<T, A1, A2, A3, A4> update, A1 arg1, A2 arg2, A3 arg3, A4 arg4) {
+    return WorkflowStartOperationUpdate.createBuilder(
+        update, new Object[] {arg1, arg2, arg3, arg4});
+  }
+
+  public static <T, A1, A2, A3, A4, A5> Builder<Void> newBuilder(
+      Functions.Proc6<T, A1, A2, A3, A4, A5> update, A1 arg1, A2 arg2, A3 arg3, A4 arg4, A5 arg5) {
+    return WorkflowStartOperationUpdate.createBuilder(
+        update, new Object[] {arg1, arg2, arg3, arg4, arg5});
+  }
+
+  public static <R> Builder<R> newBuilder(String update, Class<R> resultType, Object[] args) {
+    return new Builder<>(
+        UpdateOptions.<R>newBuilder().setResultClass(resultType).setUpdateName(update), null, args);
+  }
+
+  private static <R> Builder<R> createBuilder(Object updateFunc, Object[] args) {
```

Comment:
I'm not sure this logic should be in builder creation but rather use. Most (all?) of our builders allow you to recreate the builder from the thing being built, but this builder does not support that. There are a few ways this can be done if we want to pursue this.